Tools/ssl/multissltests.py

```diff
@@ -448,9 +448,15 @@ def _post_install_300(self):
     @property
     def short_version(self):
         """Short version for OpenSSL download URL"""
+        # strip letters, e.g. 1.1.1k -> 1.1.1
         short_version = self.version.rstrip(string.ascii_letters)
+        # remove alpha and beta suffixes, e.g. "-alpha15"
+        short_version = short_version.split("-", 1)[0]
         if short_version.startswith("0.9"):
             short_version = "0.9.x"
+        elif short_version.startswith("3.0.0"):
+            # OpenSSL 3.0.0 
+            short_version = "3.0"
```

Comment:
Better split into major, minor, patch and check for major >= 3. Otherwise you will have to rewrite this for the next major version.

Reply:
Thanks! The method now uses a regular expression.